pysat/tests/test_registry.py

```diff
@@ -41,53 +44,346 @@ def create_fake_module(full_module_name, platform, name):
+        return
+
+    def test_platform_removal_single_string(self):
+        """Test removing entire platform at once"""
+
+        # register all modules at once
+        registry.register(self.module_names)
+        # verify instantiation
+        verify_platform_name_instantiation(self.modules)
+        # verify registration
+        ensure_live_registry_updated(self.modules)
+        # verify stored update
+        ensure_updated_stored_modules(self.modules)
+        # remove them using only platform
+        for i, platform in enumerate(np.unique(self.platforms)):
+            registry.remove(platform)
+
+        # doing this one by one ensures more lines tested
+
+        return
+
+    def test_platform_name_removal_single_string(self):
+        """Test removing single platform/name at a time"""
+
+        # register all modules at once
+        registry.register(self.module_names)
+        # verify instantiation
+        verify_platform_name_instantiation(self.modules)
+        # verify registration
+        ensure_live_registry_updated(self.modules)
+        # verify stored update
+        ensure_updated_stored_modules(self.modules)
+        # remove them using only platform
+        count = 0
+        for platform, name in zip(self.platforms, self.platform_names):
+            registry.remove(platform, name)
+            # doing this one by one allows for test that all names
+            # for a platform aren't removed
+            ensure_not_in_stored_modules([self.modules[count]])
+            # test other names still present
+            if count < len(self.platforms) - 1:
+                ensure_updated_stored_modules(self.modules[count + 1:])
+            count += 1
+        # test for full removal also performed by teardown
+
+        return
+
+    def test_platform_removal_error(self):
+        """Test error raised when removing module not present"""
+
+        # remove non-registered modules using only platform
+        with pytest.raises(ValueError):
+            registry.remove(['made_up_name'])
+
+        return
+
+    def test_platform_string_removal_error(self):
+        """Test error raised when removing module not present"""
+
+        # remove non-registered modules using only platform
+        with pytest.raises(ValueError):
+            registry.remove('made_up_name')
+
+        return
+
+    def test_platform_name_removal_error(self):
+        """Test error raised when removing module not present"""
+
+        # register all modules at once
+        registry.register(self.module_names)
+
+        # remove non-registered modules using platform and name
+        with pytest.raises(ValueError):
+            registry.remove(['made_up_name'], ['made_up_name'])
+
+        # remove non-registered modules using good platform and bad name
+        with pytest.raises(ValueError):
+            registry.remove([self.platforms[0]], ['made_up_name'])
+
+        return
+
+
+class TestModuleRegistration():
+
```

Comment:
Shouldn't this have a setup and teardown?  I thought we settled on that at the last meeting.

Reply:
Fixed.